pmd-core/src/main/java/net/sourceforge/pmd/cache/CachedRuleMapper.java

```diff
@@ -9,21 +9,26 @@
 
 import net.sourceforge.pmd.Rule;
 import net.sourceforge.pmd.RuleSets;
+import net.sourceforge.pmd.annotation.InternalApi;
 
 /**
  * A mapper from rule class names to rule instances for cached rules.
  */
+@Deprecated
+@InternalApi
```

Comment:
+1, but I feel like all of this package should be internal API too. Violation caching is something that should ideally happen transparently (with user consent), and I can't see why some external client would like to manipulate the cache themselves, given how low-level the API is.

Reply:
Ok I'll do that after merging